broker/service/api/v10.py

```diff
@@ -394,6 +394,42 @@ def get_activated_cluster():
 
     global activated_cluster
     return {activated_cluster : clusters[activated_cluster]}
+
+""" Delete a done submission from the list of all
+    submissions.
+
+Raises:
+    ex.BadRequestException -- Missing parameters in request
+    ex.BadRequestException -- Trying to delete a submission that does not exists
+    ex.UnauthorizedException -- Authetication problem
+"""
+def delete_submission(submission_id, data):
+
+    check_authorization(data)
+
+    if submission_id in submissions.keys():
+        if submission_status(submission_id)['status'] in ["completed", "terminated", "error"]:
```

Comment:
`status` can also be `null`, when the submission exists in Asperathos but not in the Kubernetes cluster, either because it was not created yet (i.e. the gap between provisioning redis and visualization platforms and creating the job) or because it was deleted directly through k8s API. should you consider this state too?

Reply:
Thanks for the comment. In this method I only consider states in which for sure I know that the job was finished through Asperathos workflow. If the  `null` state is added into the condition of this method, submissions that have been created into Asperathos but didn't reach the Kubernetes Cluster can be deleted (what I think that can be a huge problem). Also, I think that Asperathos needs to have a timeout that switch the state of the job to 'error' if takes too long to reach the Kubernetes cluster. Do you have any ideas of how approach this issue?

@GabrielSVinha @liliarsampaio Any thoughts?

Reply:
I think having a timeout and going to 'error' is a better approach than considering a 'null' state, but I'm not sure if Asperathos can handle that timeout, meaning it can actually check if one of the conditions mentioned by Clenimar happened. Either way, for now, I don't think it's a blocker. Maybe we should open an issue?